### PR TITLE
[6X] Fix FNullRejecting() error while trying to process zero-placed predicate

### PR DESCRIPTION
**Pull request by InnerLife0**
When trying to do NULL rejection over zero-placed predicate, e.g. any strict function with zero arguments, we may pass NULL children array from EberEvaluate() to underlying Eber() functions. This may lead to errors, such as assertion error at EberNullOnAnyNullChild() function which used by many subclasses. As underlying Eber() functions doesn't perform any NULL check, we need to pass an empty array to them in case we working with zero-placed predicate.
Another option was to return EberAny from EberEvaluate() in case we dealing with zero arity, but there is underlying CScalarConst::Eber() function which should be called and which do it's stuff ignoring incoming argument at all. Here I decided to keep current solution as it's clean.

A little highlighting of function folding. If you'll remove topmost select from any of new tests, you'll not get an error because strict function will be folded, and thus, became Const. Starting from 4cf47434a35df73dd456b6648c5f40ee377b5845 such functions under topmost select not folded, because such select has no parse->rtable. At the other hand, Postgres planner, on which ORCA fell back before current patch, can do folding because of separate calls to eval_const_expressions() from subplans

### CODE REVIEW

File: src/test/regress/expected/gporca_optimizer.out [hunk 1]
```diff
@@ -14515,6 +14515,59 @@ select * from part_table13 where c=7;
...
+ Optimizer: Pivotal Optimizer (GPORCA)
+(15 rows)
+
+-- Optional, but let's check we get same result for both, folded and
+-- not folded now() frunction.
+SELECT (
+    SELECT count(*) cnt
+    FROM join_null_rej1 t1
+    LEFT JOIN join_null_rej2 t2 ON t1.i = t2.i
+    WHERE t2.i < join_null_rej_func()
```
Comment (bhuvnesh2703) [inline src/test/regress/expected/gporca_optimizer.out:14564]: - typo frunction
- `also folded and not folded now()` looks like this statement need to be corrected to represent the join_null_rej_func func.

Comment (InnerLife0) [reply src/test/regress/expected/gporca_optimizer.out:14564]: Thanks! Fixed.

**Review by chrishajas — APPROVED** [review]

**Review by bhuvnesh2703 — APPROVED** [review]

**Comment by bhuvnesh2703** [comment]
Thank your @InnerLife0 for the contribution. I will take it to the Review Committee to approve / disapprove it for 6X.

**Comment by InnerLife0** [comment]
I'll squash it once again if all is OK.

**Comment by bhuvnesh2703** [comment]
Got Approval. Please squash it, and i can merge afterwards.

**Comment by InnerLife0** [comment]
Squashed and CI is OK.

**Pull request merged**